### ISSUE TITLE
[Performance] Ship loading demands too much of gpu

### ISSUE THREAD

**Issue by Uncle-Ulty** [tsunamayo/Starship-EVO#5438]
### Short Description of the issue:
Loading a big ship makes the FPS drop too much.
...

### Steps to reproduce the issue:

-Start a new world
-place a new entity and spawn a ship: TEC-DEFENSOR

The ship takes some time to load. It starts as invisible and than loads piece by piece

I don't remember this ship being so performance burden, but a player warned me in the workshop about the issue. So I don't know if it's a bug or something else.

cheers!

![image](https://github.com/tsunamayo/Starship-EVO/assets/73253472/87158159-e30c-49fa-b67a-36dde1deaeaf)

### Environment
- Starship EVO version: 24w03a
- Graphics Card: gtx1660

### Additional information:

_(Screenshots, Videos, Error logs, Related issues)_

https://youtu.be/7m3etiFHsr4

Please also include the `Player.log` file. You can find the log file here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`
[Player.log](https://github.com/tsunamayo/Starship-EVO/files/14038359/Player.log)
[Teste de updates.zip](https://github.com/tsunamayo/Starship-EVO/files/14038413/Teste.de.updates.zip)

If you issue concerns a particular design please include any blueprint that could help to reproduce the issue.
Blueprint folder is over at:
[TEC - Defensor Frigate.zip](https://github.com/tsunamayo/Starship-EVO/files/14038435/TEC.-.Defensor.Frigate.zip)

`%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\Save_Data\Blueprints\`
You will need to zip the folder to upload it. 
To do so Right click and choose [Send to -> compressed zip file] then drag the .zip onto your report to upload it.

**Comment by tsunamayo** [comment]
@Uncle-Ulty Hi, this is ship is glorious ^^.
Is it just a loading problem? If yes then no big deal, I changed the loading logic a few update back so that wreck spawning does not trigger hiccup. Large ship will take a few seconds to load (the games now load 500 bricks per frame).
Cheers

**Comment by Uncle-Ulty** [comment]
I see.

Yeah, I noticed the loading isn't freezing the game any longer. But after the loading process my game runs at 40 FPS. I'm kinda worried cuz I don't remember this ship being so performance burden.  I've been building something bigger and I fear it will cause more FPS drop :(.

**Comment by tsunamayo** [comment]
@Uncle-Ulty Hum no it should not run less smooth, it was fine when I loaded the ship. Can you reproduce that all the time? Where you sure it was running smoother before? Thanks

**Comment by Uncle-Ulty** [comment]
Yeah, I can reproduce it all the time. Back when I was building it in Aug 2022, I don't remember my fps dropping below 60fps.

Maybe my GTX1660 is too old for this game? idk

**Comment by tsunamayo** [comment]
@Uncle-Ulty Well it is not supposed to run worse than before, and GTX1660 should be good enough. Can we make sure you are GPU bound? Can you get near your ship, and look at the empty void and tell me the FPS you do? Thanks

**Comment by Uncle-Ulty** [comment]
screen record:
https://youtu.be/Cn_SmVI77tc

Enable shadow: on
Looking away: 60 fps (normal)
Looking to Defensor: 40 fps

Enable shadow: off:
Looking away 60 fps
Looking to defensor 60 fps

It seems the "enable shadow" option consumes a lot of performance... Does the game process the shadow of the entire ship at once, or just the visible shadow?
